src/Analyzers/VisualBasic/CodeFixes/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider.vb

```diff
@@ -0,0 +1,73 @@
+        Public Sub New()
+        End Sub
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) = ImmutableArray.Create(IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.CodeStyle
+
+        Public Overrides Async Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            Dim root = Await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(False)
+            For Each diagnostic In context.Diagnostics
+                Dim node = root.FindNode(diagnostic.Location.SourceSpan)
+                If TypeOf node Is ArgumentListSyntax Then ' The analyzer reported parentheses that should be removed.
+                    ' Remove the parentheses.
+                    context.RegisterCodeFix(New MyCodeAction(
+                        VisualBasicAnalyzersResources.Add_invocation_parentheses, ' TODO: Different message here.
+                        Function(ct) FixAsync(context.Document, diagnostic, ct)),
+                        diagnostic)
+                ElseIf TypeOf node Is IdentifierNameSyntax Then ' The analyzer reported absence of parentheses that should be added.
+                    ' Add the parentheses.
+                    context.RegisterCodeFix(New MyCodeAction(
+                        VisualBasicAnalyzersResources.Add_invocation_parentheses,
+                        Function(ct) FixAsync(context.Document, diagnostic, ct)),
+                        diagnostic)
+                End If
+            Next
+        End Function
+
+        Protected Overrides Async Function FixAllAsync(document As Document, diagnostics As ImmutableArray(Of Diagnostic), editor As SyntaxEditor, cancellationToken As CancellationToken) As Task
+            Dim root = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
+            For Each diagnostic In diagnostics
+                Dim node = root.FindNode(diagnostic.AdditionalLocations(0).SourceSpan)
+                If TypeOf node IsNot InvocationExpressionSyntax Then
+                    Return
+                End If
+                ' Should I get editorconfig options here? and how?
+                ' Can I rely on the following logic without checking the editorconfig?
+                '     - The analyzer reports diagnostic if and only if an action needs to be taken.
+                '     - So, if parentheses exist, remove them.
+                '     - If they don't exist, add them.
+            Next
+        End Function
+
+        Private Class MyCodeAction
+            Inherits CustomCodeActions.DocumentChangeAction
+
+            Friend Sub New(title As String, createChangedDocument As Func(Of CancellationToken, Task(Of Document)))
+                MyBase.New(title, createChangedDocument)
```

Comment:
```suggestion
                MyBase.New(title, createChangedDocument, title)
```

Reply:
Does the absence of the title as equivalence key causes any problems?
I believe I didn't have it the RemoveUnnecessaryByVal analyzer, so that might need to get updated too.